envoy/network/listen_socket.h

```diff
@@ -26,7 +25,7 @@ namespace Network {
  * TODO(jrajahalme): Hide internals (e.g., fd) from listener filters by providing callbacks filters
  * may need (set/getsockopt(), peek(), recv(), etc.)
  */
-class ConnectionSocket : public virtual Socket, public virtual ScopeTrackedObject {
```

Comment:
I'm curious why do we need this change?

Reply:
This is used to avoid cycle dependency. And socket doesn't need to inherit from scope tracked object.

Reply:
Make sense.